lib/astarte_data_access/database.ex

```diff
@@ -19,12 +19,13 @@
 defmodule Astarte.DataAccess.Database do
   require Logger
 
-  @spec connect() :: {:ok, :cqerl.client()} | {:error, atom}
-  def connect() do
-    with {:nodes, nodes} when is_list(nodes) <-
-           {:nodes, Application.get_env(:cqerl, :cassandra_nodes)},
+  @spec connect(realm: String.t(), cassandra_nodes: list) ::
+          {:ok, :cqerl.client()} | {:error, atom}
+  def connect(opts \\ []) do
+    with {:nodes, nodes} when is_list(nodes) <- get_nodes(opts),
+         {:keyspace, keyspace} <- get_keyspace(opts),
```

Comment:
this is actually `get_client_opts` since you're effectively returning the opts keyword list

Reply:
Also, no need to use `<-` since it always succeeds (either you have opts with a keyspace or you have empty opts), just do
```elixir
    client_opts = get_client_opts(opts),
    ...
```